### PR TITLE
Restored Maven 3.0.5 formula

### PR DESCRIPTION
**Pull request by pkuczynski**

### CODE REVIEW

File: maven30.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+class Maven30 < Formula
```
Comment (MikeMcQuaid) [inline maven30.rb]: We won't accept this because of https://github.com/Homebrew/homebrew-versions#acceptable-formulae

File: maven31.rb [hunk 2]
```diff
@@ -6,7 +6,10 @@ class Maven31 < Formula
 
   depends_on :java
 
-  conflicts_with "maven", :because => "Differing versions of same formula"
+  conflicts_with "maven", :because => "maven31 is older versions of this formula"
+  conflicts_with "maven30", :because => "maven31 is newer versions of this formula"
```
Comment (MikeMcQuaid) [inline maven31.rb]: Remove this line.

File: maven31.rb [hunk 3]
```diff
@@ -6,7 +6,10 @@ class Maven31 < Formula
 
   depends_on :java
 
-  conflicts_with "maven", :because => "Differing versions of same formula"
+  conflicts_with "maven", :because => "maven31 is older versions of this formula"
```
Comment (MikeMcQuaid) [inline maven31.rb]: Leave this option description as-is.

File: maven32.rb [hunk 4]
```diff
@@ -7,8 +7,10 @@ class Maven32 < Formula
 
   depends_on :java
 
-  conflicts_with "maven", :because => "Differing versions of same formula"
```
Comment (MikeMcQuaid) [inline maven32.rb]: Leave this option description as-is.

File: maven32.rb [hunk 5]
```diff
@@ -7,8 +7,10 @@ class Maven32 < Formula
...
-  conflicts_with "mvnvm", :because => "Also installs a 'mvn' executable"
+  conflicts_with "maven", :because => "maven32 is older versions of this formula"
+  conflicts_with "maven30", :because => "maven32 is newer versions of this formula"
```
Comment (MikeMcQuaid) [inline maven32.rb]: Remove this line.

File: maven30.rb (hunk 1 above)
Comment (pkuczynski) [inline maven30.rb]: Could you please elaborate which rule this PR is violating?

Comment (MikeMcQuaid) [inline maven30.rb]: Versions formulae must not exceed +/-2 major/minor (not patch) versions from the current stable release.

Comment (pkuczynski) [inline maven30.rb]: Well, it says +/-2, and Maven 3.0.5 released in 2013 is still used for many projects. Maybe you could bend this rule and accept it? Looking at http://maven.apache.org/docs/history.html it is still on the list of supported major versions...

Comment (MikeMcQuaid) [inline maven30.rb]: This is -3. Sorry, we won't accept this. Please consider [creating your own tap](https://github.com/Homebrew/brew/blob/master/share/doc/homebrew/How-to-Create-and-Maintain-a-Tap.md#how-to-create-and-maintain-a-tap) instead. Thanks!

Comment (pkuczynski) [inline maven30.rb]: Your like gives 404. 

I accept your approach, although I do not understand it. You can close my pull request...

Comment (MikeMcQuaid) [inline maven30.rb]: My apologies, it's here: https://github.com/Homebrew/brew/blob/master/docs/How-to-Create-and-Maintain-a-Tap.md#how-to-create-and-maintain-a-tap

We can't maintain old versions indefinitely for both security and time reasons, sorry.

**Pull request closed without merging**